cpp/unreal_plugins/SpCore/Source/SpCore/StableNameComponent.h

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#pragma once
+
+#include <Components/SceneComponent.h>
+#include <Containers/UnrealString.h> // FString
+#include <UObject/ObjectMacros.h>    // GENERATED_BODY, UCLASS
+
+#include "StableNameComponent.generated.h"
+
+// We need meta=(BlueprintSpawnableComponent) for the component to show up when using the "+Add" button in the editor.
+UCLASS(ClassGroup="SPEAR", HideCategories=(Rendering, Tags, Activation, Cooking, Physics, LOD, AssetUserData, Collision), meta=(BlueprintSpawnableComponent))
+class SPCORE_API UStableNameComponent : public USceneComponent
+{
+    GENERATED_BODY()
+public:
+    UStableNameComponent();
+    ~UStableNameComponent();
+
+    // If we're in the editor, then set StableName when the component is created or when the editor starts.
+    #if WITH_EDITOR // defined in an auto-generated header
+        void OnComponentCreated() override;
+        void PostLoad() override;
+    #endif
+
+    UPROPERTY(VisibleAnywhere, Category = "SPEAR", DisplayName = "Stable Name");
+    FString StableName;
+
+    // If we're in the editor, provide a public function to set StableName. This is useful, e.g., when the
+    // owning actor is renamed or moved in the World Outliner.
+#if WITH_EDITOR // defined in an auto-generated header
+        void update();
```

Comment:
Should this function's name be `updateStableName()` rather than `update()`, as the former conveys the purpose of this function more clearly.

Reply:
9 times out of 10 I would agree with you. `update()` is usually too vague of a method name. But in this case I think it's fine, because any usage of this method would say something like `stable_name_component->update()`, which is the same level of information content as `stable_name_component->updateStableName()`, but is shorter.